ecl/hql/hqlexpr.cpp

```diff
@@ -924,6 +924,16 @@ static void setDefinitionText(IPropertyTree * target, const char * prop, IFileCo
     target->setProp("@sourcePath", str(sourcePath));
 }
 
+IPropertyTree * HqlParseContext::beginMetaSource(IFileContents * contents)
+{
+    ISourcePath * sourcePath = contents->querySourcePath();
+
+    IPropertyTree * attr = createPTree("Source", ipt_fast);
+    attr->setProp("@sourcePath", str(sourcePath));
+    metaState.nesting.append(*attr);
```

Comment:
The definition of nesting changed from
    PointerArrayOf<IPropertyTree> nesting;
to
    IArrayOf<IPropertyTree> nesting;
so it now needs references rather than pointers added to the array.